src/common/catalog-entities/kubernetes-cluster.ts

```diff
@@ -0,0 +1,105 @@
+import { EventEmitter } from "events";
+import { observable } from "mobx";
+import { catalogCategoryRegistry } from "../catalog-category-registry";
+import { CatalogCategory, CatalogEntity, CatalogEntityActionContext, CatalogEntityContextMenuContext, CatalogEntityData, CatalogEntityMetadata, CatalogEntityStatus } from "../catalog-entity";
+import { clusterDisconnectHandler } from "../cluster-ipc";
+import { clusterStore } from "../cluster-store";
+import { requestMain } from "../ipc";
+
+export type KubernetesClusterSpec = {
+  kubeconfigPath: string;
+  kubeconfigContext: string;
+};
+
+export interface KubernetesClusterStatus extends CatalogEntityStatus {
+  phase: "connected" | "disconnected";
+}
+
+export class KubernetesCluster implements CatalogEntity {
+  public readonly apiVersion = "entity.k8slens.dev/v1alpha1";
+  public readonly kind = "KubernetesCluster";
+  @observable public metadata: CatalogEntityMetadata;
+  @observable public status: KubernetesClusterStatus;
+  @observable public spec: KubernetesClusterSpec;
+
+  constructor(data: CatalogEntityData) {
+    this.metadata = data.metadata;
+    this.status = data.status as KubernetesClusterStatus;
+    this.spec = data.spec as KubernetesClusterSpec;
+  }
+
+  getId() {
+    return this.metadata.uid;
+  }
+
+  getName() {
+    return this.metadata.name;
+  }
+
+  async onRun(context: CatalogEntityActionContext) {
+    context.navigate(`/cluster/${this.metadata.uid}`);
+  }
+
+  async onDetailsOpen() {
+    //
+  }
+
+  async onContextMenuOpen(context: CatalogEntityContextMenuContext) {
+    context.menuItems = [
+      {
+        icon: "settings",
+        title: "Settings",
+        onClick: async () => context.navigate(`/cluster/${this.metadata.uid}/settings`)
+      },
+      {
+        icon: "delete",
+        title: "Delete",
+        onClick: async () => clusterStore.removeById(this.metadata.uid),
+        confirm: {
+          message: `Remove Kubernetes Cluster "${this.metadata.name} from Lens?`
+        }
```

Comment:
Yes, kind of temporary. We need to figure out how multiple sources can introduce same menu action without duplicate menu items. For example `source=gke-sync` should be able to override `delete` because it makes no sense to remove local state from Lens.

I would prefer that we solve this problem in a follow-up PR.